avalon/tools/loader/model.py

```diff
@@ -141,17 +150,72 @@ def set_version(self, index, version):
             "step": version_data.get("step", None)
         })
 
-    def refresh(self):
+    def fetch_subset_and_version(self):
+        """Query all subsets and latest versions from aggregation
+
+        (NOTE) The returned version documents are NOT the real version
+            document, it's generated from the MongoDB's aggregation so
+            some of the first level field may not be presented.
+
+        """
+        def _fetch():
+            _subsets = list()
+            _ids = list()
+            for subset in io.find({"type": "subset",
+                                   "parent": self._asset_id}):
+                if self._doc_fetching_stop:
+                    return
+                _subsets.append(subset)
+                _ids.append(subset["_id"])
+
+            _pipeline = [
```

Comment:
Would be great if there is (e.g. in avalon.lib) function for this pipeline to be usable in multiple places.
```
def last_version_pipeline(subset_ids):
    return [
        # Find all versions of those subsets
        {"$match": {"type": "version", "parent": {"$in": subset_ids}}},
        ...
    ]
```

But it is only nice to have, because it is not used in other places yet.

Reply:
Maybe it could be in the `avalon.io` module. 🤔